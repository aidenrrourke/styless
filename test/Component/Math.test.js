import React from 'react';
import renderer from 'react-test-renderer';
import styled from 'styled-components';
import 'jest-styled-components';

test('Rounds up to the next highest integer.', () => {
    const Div = styled.div`
        ceil: ceil(2.4);
        ceil: ceil(@value);
        ceil: ceil(2.4px);
        ceil: ceil(@value2);
	`;
    expect(renderer.create(<Div value={2.4} value2="2.4px"/>).toJSON()).toMatchSnapshot();
});

test('Rounds down to the next lowest integer.', () => {
    const Div = styled.div`
        floor: floor(2.6);
        floor: floor(@value);
        floor: floor(2.6em);
        floor: floor(@value2);
	`;
    expect(renderer.create(<Div value={2.6} value2="2.6em"/>).toJSON()).toMatchSnapshot();
});

test('Converts a floating point number into a percentage string.', () => {
    const Div = styled.div`
        p: percentage(0.5);
        p: percentage(0.5em);
        p: percentage(@value);
	`;
    expect(renderer.create(<Div value={0.5}/>).toJSON()).toMatchSnapshot();
});

test('Applies rounding.', () => {
    const Div = styled.div`
        round: round(1.67);
        round: round(@value);
        round: round(1.67px);
        round: round(@value2);
	`;
    expect(renderer.create(<Div value={1.67} value2="1.67px"/>).toJSON()).toMatchSnapshot();
});

test('Calculates square root of a number. Keeps units as they are.', () => {
    const Div = styled.div`
        sqrt: sqrt(25cm);
        sqrt: sqrt(@value);
	`;
    expect(renderer.create(<Div value="25cm"/>).toJSON()).toMatchSnapshot();
});