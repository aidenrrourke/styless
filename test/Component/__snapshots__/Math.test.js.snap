// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Applies rounding. 1`] = `
.c0 {
  round: 2;
  round: 2;
  round: 2px;
  round: 2px;
}

<div
  className="c0"
  value={1.67}
/>
`;

exports[`Calculates square root of a number. Keeps units as they are. 1`] = `
.c0 {
  sqrt: 5cm;
  sqrt: 5cm;
}

<div
  className="c0"
  value="25cm"
/>
`;

exports[`Converts a floating point number into a percentage string. 1`] = `
.c0 {
  p: 50%;
  p: 50%;
  p: 50%;
}

<div
  className="c0"
  value={0.5}
/>
`;

exports[`Rounds down to the next lowest integer. 1`] = `
.c0 {
  floor: 2;
  floor: 2;
  floor: 2em;
  floor: 2em;
}

<div
  className="c0"
  value={2.6}
/>
`;

exports[`Rounds up to the next highest integer. 1`] = `
.c0 {
  ceil: 3;
  ceil: 3;
  ceil: 3px;
  ceil: 3px;
}

<div
  className="c0"
  value={2.4}
/>
`;
